conda_devenv/devenv.py

```diff
@@ -490,35 +514,34 @@ def get_env_name(args, output_filename, conda_yaml_dict=None):
     if conda_yaml_dict is None:
         import yaml
 
-        with open(output_filename, "r") as stream:
+        with open(output_filename) as stream:
             conda_yaml_dict = yaml.safe_load(stream)
 
     return conda_yaml_dict["name"]
 
 
-def _get_envs_dirs_from_conda():
+def _get_envs_dirs_from_conda() -> Sequence[Path]:
     from conda.base.context import context
 
-    return context.envs_dirs
+    return [Path(x) for x in context.envs_dirs]
 
 
-def get_env_directory(env_name):
+def get_env_directory(env_name: str) -> Path | None:
     """
-    :rtype: Optional[str]
-    :return: The environment path if the enviromment exists.
+    The environment path if the enviromment exists.
     """
     envs_dirs = _get_envs_dirs_from_conda()
 
     for directory in envs_dirs:
         env = os.path.join(directory, env_name)
         conda_meta_dir = os.path.join(env, "conda-meta")
         if os.path.isdir(conda_meta_dir):
-            return os.path.normpath(env)
+            return Path(os.path.normpath(env))
```

Comment:
Is this a possible breaking change on the public API?
For example, if someone took the output and parsed it as a `str`, the code will now break.
I'm not against this change, though.

Reply:
Oh I did not realize we had a public API.

But even if we did, should be 1 or 2 projects at most, so I prefer to update them. I will mention in the changelog.